backend/core/src/main/java/org/sonarsource/sonarlint/core/BindingClueProvider.java

```diff
@@ -141,38 +170,57 @@ private Set<String> matchConnections(BindingClue bindingClue, Set<String> eligib
   }
 
   @CheckForNull
-  private static ScannerProperties extractScannerProperties(ClientFile matchedFile) {
-    LOG.debug("Extracting scanner properties from {}", matchedFile);
-    var properties = new Properties();
+  private static ConnectionProperties  extractSonarLintConfiguration(ClientFile sonarLintConfigurationFile) {
     try {
-      properties.load(new StringReader(matchedFile.getContent()));
-    } catch (Exception e) {
-      LOG.error("Unable to parse content of file '{}'", matchedFile, e);
+      var configuration = new Gson().fromJson(sonarLintConfigurationFile.getContent(), JsonObject.class);
+      var projectKey = configuration.get("projectKey");
+      var organization = configuration.get("organization");
+      var serverUrl = configuration.get("serverUrl");
+      return new ConnectionProperties(projectKey != null ? projectKey.getAsString() : null,
+        organization != null ? organization.getAsString() : null,
+        serverUrl != null ? serverUrl.getAsString() : null);
+    } catch (JsonSyntaxException e) {
       return null;
     }
-    return new ScannerProperties(getAndTrim(properties, "sonar.projectKey"), getAndTrim(properties, "sonar.organization"),
-      getAndTrim(properties, "sonar.host.url"));
+  }
+
+  @CheckForNull
+  private static ConnectionProperties extractConnectionProperties(ClientFile matchedFile) {
+    LOG.debug("Extracting scanner properties from {}", matchedFile);
+    if (matchedFile.isSonarlintConfigurationFile()) {
+      return extractSonarLintConfiguration(matchedFile);
+    } else {
+      var properties = new Properties();
+      try {
+        properties.load(new StringReader(matchedFile.getContent()));
+      } catch (Exception e) {
+        LOG.error("Unable to parse content of file '{}'", matchedFile, e);
+        return null;
+      }
+      return new ConnectionProperties(getAndTrim(properties, "sonar.projectKey"), getAndTrim(properties, "sonar.organization"),
+        getAndTrim(properties, "sonar.host.url"));
+    }
   }
 
   @CheckForNull
   private static String getAndTrim(Properties properties, String key) {
     return trimToNull(properties.getProperty(key));
   }
 
-  private static class ScannerProperties {
+  private static class ConnectionProperties {
     private final String projectKey;
     private final String organization;
     private final String serverUrl;
 
-    private ScannerProperties(@Nullable String projectKey, @Nullable String organization, @Nullable String serverUrl) {
+    private ConnectionProperties(@Nullable String projectKey, @Nullable String organization, @Nullable String serverUrl) {
```

Comment:
Not sure it's a "connection" properties, as there is a project key involved. Maybe `BindingProperties`?